src/coreclr/jit/promotionliveness.cpp

```diff
@@ -688,8 +688,19 @@ void PromotionLiveness::FillInLiveness(BitVec& life, BitVec volatileVars, GenTre
         if (lcl->OperIs(GT_LCL_ADDR))
         {
             // Retbuf -- these are definitions but we do not know of how much.
-            // We never mark them as dead and we never treat them as killing anything.
-            assert(isDef);
+            // We never treat them as killing anything, but we do store liveness information for them.
+            BitVecTraits aggTraits(1 + (unsigned)agg->Replacements.size(), m_compiler);
+            BitVec       aggDeaths(BitVecOps::MakeEmpty(&aggTraits));
+            // Copy preexisting liveness information.
+            for (size_t i = 0; i <= agg->Replacements.size(); i++)
+            {
+                unsigned varIndex = baseIndex + (unsigned)i;
+                if (!BitVecOps::IsMember(m_bvTraits, life, varIndex))
+                {
+                    BitVecOps::AddElemD(&aggTraits, aggDeaths, (unsigned)i);
+                }
+            }
+            m_aggDeaths.Set(lcl, aggDeaths);
```

Comment:
We need this liveness here now to know what fields we need to mark for readback when a physically promoted struct is passed as the retbuf.